src/Storages/StorageBuffer.cpp

```diff
@@ -883,8 +854,6 @@ void StorageBuffer::flushBuffer(Buffer & buffer, bool check_thresholds, bool loc
     try
     {
         writeBlockToDestination(block_to_write, DatabaseCatalog::instance().tryGetTable(destination_id, getContext()));
-        if (reset_block_structure)
-            buffer.data.clear();
```

Comment:
Why is it removed?

Reply:
This is not required anymore, since Block (buffer.data) is empty, due to this line https://github.com/ClickHouse/ClickHouse/pull/33324/files#diff-c2fcfac3d220fdad23241404d8ed9a2797aadbe40a45ab16dfce3e9f67208689L836